src/cargo/core/compiler/fingerprint.rs

```diff
@@ -342,6 +344,9 @@ use super::job::{
 };
 use super::{BuildContext, Context, FileFlavor, Unit};
 
+type FileSize = u32;
```

Comment:
Files may be bigger than 4GB. For example servo's script is larger than 4GB when debuginfo is enabled I think.

Reply:
I should have been clearer, I was aiming for the 'size' to be the lower 32bits of the size as that should be almost as good as the full number for this purpose. Whether that optimisation is a good idea, idk but will use a little less memory.

Reply:
@gilescope It shouldn't be substantially slower to hash in the full size, since this will be a hash rather than storing all the individual data points that go into the hash.

Reply:
Agreed - I think on the whole it's less confusing if the size is the file size rather than the lowest 32 bits of the file size.